### PR TITLE
Break out GTest into its own package to fix windows CI build failures

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
@marip8 For some reason the current method was causing all of my windows CI to fail. This reverts the changes, but adds gtest as another package which should achieve the same results but fix my windows CI failures.

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@marip8 Do you mind reviewing?

File: README.rst
Comment (marip8) [inline README.rst:145]:
```suggestion
This CMake macro calls `find_package(GTest REQUIRED)` and checks for the `GTest::GTest` and `GTest::Main` targets. If the targets are missing it will create the targets using the CMake variables.
```

File: gtest/CMakeLists.txt
Comment (marip8) [inline gtest/CMakeLists.txt:16]: Does the entire `ExternalProject_Add` command need to be duplicated between the Linux and MSVC if statement? It looks like almost everything is the same except the path and extensions of the libraries.

```cmake
# Common configuration
ExternalProject_Add(GTest
...
)
file(...)
set(GTEST_INCLUDE_DIRS ...)

# Compiler specific configuration
if(GNU / CLANG ...)
  set(GTEST_LIBRARIES ...)
  set(GTEST_MAIN_LIBRARIES ...)
elseif(MSVC)
  set(GTEST_LIBRARIES ...)
  set(GTEST_MAIN_LIBRARIES ...)
endif()
```

File: gtest/package.xml
Comment (marip8) [inline gtest/package.xml:7]:
```suggestion
  <license>Apache 2.0</license>
```

**Review by marip8 — CHANGES_REQUESTED** [review]

File: gtest/package.xml
Comment (Levi-Armstrong) [reply gtest/package.xml]: Fixed

File: gtest/CMakeLists.txt
Comment (Levi-Armstrong) [reply gtest/CMakeLists.txt]: updated

**Comment by Levi-Armstrong** [comment]
@marip8 How does it look now? I will squash merge when approved.

**Comment by Levi-Armstrong** [comment]
I am currently testing this [here](https://github.com/ros-industrial-consortium/tesseract/pull/461)

File: README.rst
Comment (marip8) [inline README.rst:144]: I think RST actually requires two back-ticks

```suggestion
This CMake macro calls ``find_package(GTest REQUIRED)`` and checks for the ``GTest::GTest`` and ``GTest::Main`` targets. If the targets are missing it will create the targets using the CMake variables.
```

**Comment by marip8** [comment]
Other than the README formatting, I think this looks good

**Comment by Levi-Armstrong** [comment]
@marip8 Thanks, I will wait for confirmation from the Tesseract PR before squash merging.

**Pull request merged**
